offer-manager-domain/src/main/groovy/life/qbic/business/offers/Offer.groovy

```diff
@@ -277,14 +279,16 @@ class Offer {
     }
 
     private BigDecimal calculateItemNet(ProductItem item) {
-        BigDecimal unitPrice = selectedCustomerAffiliation.category == AffiliationCategory.INTERNAL ? item.product.internalUnitPrice : item.product.externalUnitPrice
-        return unitPrice * item.quantity
+        return getUnitPrice(item) * item.quantity
     }
 
     private double calculateItemOverhead(ProductItem item) {
         return (calculateItemNet(item) - item.quantityDiscount) * overhead
     }
 
+    private BigDecimal getUnitPrice(ProductItem item){
```

Comment:
```suggestion
    private BigDecimal determineUnitPrice(ProductItem item){
```
I would not phrase it as a getter method since you are performing logic here.